collector/database/classes/collector.php

```diff
@@ -72,4 +88,40 @@ public function get_metrics($metricnames = null, $since = false): array {
                ORDER BY time asc";
         return $DB->get_records_sql($sql, $params);
     }
+
+    /**
+     * Records retrieved data in collector.
+     *
+     * @param  \tool_cloudmetrics\metric\base $metricclass Class representing metric.
+     * @param  array $data Array of metric items.
+     */
+    public function record_saved_metrics(\tool_cloudmetrics\metric\base $metricclass, array $metricitems = []) {
+        global $DB;
+
+        $maxbackdate = explode('-', $metricclass->get_range_retrieved())[0];
+        $transaction = $DB->start_delegated_transaction();
+        // Sets what data has been sent to collector.
+        $metricclass->set_data_sent_config();
```

Comment:
you should avoid setting config inside a transaction, if it rolls back you can end up with the db and the MUC out of sync because the MUC doesn't roll back. I'd do it after it has committed

Reply:
Fixed.